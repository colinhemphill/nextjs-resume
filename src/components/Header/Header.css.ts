import { atoms } from '../../strum-design-system/sprinkles.css';

export const headerStyle = atoms({
  backgroundColor: { darkMode: 'black', lightMode: 'primary' },
  paddingY: 8,
});
