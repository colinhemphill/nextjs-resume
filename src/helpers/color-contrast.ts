import config from '../../edit-me/config/resume-config';
import { AccentColorsBright, ResumeConfig } from '../../edit-me/types/config';

const isBrightColor = (
  color: ResumeConfig['accentColor'],
): color is AccentColorsBright => {
  const brightColors = Object.values(AccentColorsBright) as string[];
  return brightColors.includes(color);
};

export const contrastColor = isBrightColor(config.accentColor)
  ? '#000'
  : '#fff';
