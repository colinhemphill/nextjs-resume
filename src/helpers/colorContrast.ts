import { AccentColorsBright, ResumeConfig } from '../../edit-me/types/Config';
import config from '../../edit-me/config/resumeConfig';

const isBrightColor = (
  color: ResumeConfig['accentColor'],
): color is AccentColorsBright => {
  const brightColors = Object.values(AccentColorsBright) as Array<string>;
  return brightColors.includes(color);
};

export const contrastColor = isBrightColor(config.accentColor)
  ? '#000'
  : '#fff';
