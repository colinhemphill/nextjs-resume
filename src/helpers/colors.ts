import * as colors from '@radix-ui/colors';
import { Theme } from '../../edit-me/types/Config';
import resumeConfig from '../../edit-me/config/resumeConfig';

const defaultTheme = resumeConfig.imageTheme;
const configAccent = resumeConfig.accentColor;
const configNeutral = resumeConfig.neutralColor;

/**
 * Get a Radix accent color based on your resume config
 * @param shade A number from 1 to 12 representing the Radix Colors shade
 * @returns The accent color string
 */
export const getAccentColor = (shade: number, theme = defaultTheme) => {
  const accent =
    colors[theme === Theme.Dark ? `${configAccent}Dark` : configAccent];
  return accent[`${configAccent}${shade}`];
};

/**
 * Get a Radix neutral color based on your resume config
 * @param shade A number from 1 to 12 representing the Radix Colors shade
 * @returns The neutral color string
 */
export const getNeutralColor = (shade: number, theme = defaultTheme) => {
  const neutral =
    colors[theme === Theme.Dark ? `${configNeutral}Dark` : configNeutral];
  return neutral[`${configNeutral}${shade}`];
};
