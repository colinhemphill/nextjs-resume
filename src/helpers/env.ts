export const vercelURL = process.env.NEXT_PUBLIC_VERCEL_URL;
export const dev = process.env.NODE_ENV === 'development';
export const protocol = dev ? 'http' : 'https';
