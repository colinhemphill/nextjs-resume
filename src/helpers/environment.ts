export const deployURL =
  process.env.DEPLOY_URL ??
  process.env.URL ??
  process.env.NEXT_PUBLIC_VERCEL_URL;
export const development = process.env.NODE_ENV === 'development';
export const protocol = development ? 'http' : 'https';
