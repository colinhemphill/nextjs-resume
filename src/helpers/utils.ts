import { personal } from '@content';

export const fullName = `${personal.givenName} ${personal.familyName}`;
