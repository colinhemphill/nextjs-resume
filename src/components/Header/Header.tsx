import { personal } from '@content';
import React from 'react';
import { fullName } from '../../helpers/utils';
import { Heading } from '../Heading/Heading';
import PDFDownloadButton from '../PDF/PDFDownloadButton';

interface HeaderProps {
  secret?: string;
}

export const Header: React.FC<HeaderProps> = ({ secret }) => {
  return (
    <div className="mb-12 border-b-2 border-neutral-4 py-12">
      <div className="container">
        <div className="flex flex-col items-center gap-6 text-center md:flex-row md:text-left">
          <div className="flex flex-1 flex-col gap-2">
            <Heading level={1}>{fullName}</Heading>
            <Heading color="neutralSubtle" level={2}>
              {personal.title}
            </Heading>
          </div>
          <PDFDownloadButton secret={secret} />
        </div>
      </div>
    </div>
  );
};
