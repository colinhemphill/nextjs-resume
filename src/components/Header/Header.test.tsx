import { personal } from '@content';
import { render, screen } from '@test-utils';
import { fullName } from 'src/helpers/utils';
import { describe, expect, test } from 'vitest';
import { Header } from './Header';

describe('<Header />', () => {
  test('Snapshot', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });

  test('Renders headings and PDF button', () => {
    render(<Header />);

    expect(
      screen.getByRole('heading', { level: 1, name: fullName }),
    ).toBeDefined();
    expect(
      screen.getByRole('heading', { level: 2, name: personal.title }),
    ).toBeDefined();
    expect(
      screen.getByRole('link', { name: /view or download pdf/i }),
    ).toBeDefined();
  });
});
