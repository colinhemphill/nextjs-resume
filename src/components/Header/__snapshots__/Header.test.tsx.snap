// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Header /> > Snapshot 1`] = `
<div>
  <div
    class="mb-12 border-b-2 border-neutral-4 py-12"
  >
    <div
      class="container"
    >
      <div
        class="flex flex-col items-center gap-6 text-center md:flex-row md:text-left"
      >
        <div
          class="flex flex-1 flex-col gap-2"
        >
          <h1
            class="font-bold text-neutral-12 text-4xl md:text-5xl"
          >
            Colin Hemphill
          </h1>
          <h2
            class="font-bold text-neutral-11 text-2xl md:text-3xl"
          >
            Official Job Title
          </h2>
        </div>
        <a
          class="flex items-center justify-center gap-2 rounded-md border-2 border-accent-7 bg-accent-1 font-medium text-accent-11 no-underline outline-none transition hover:border-accent-8 focus:border-transparent focus:bg-accent-9 focus:text-accentContrast focus:ring-4 focus:ring-accent-6 hover:focus:border-transparent text-lg py-3 px-8"
          href="/api/pdf"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-file-pdf fa-lg "
            data-icon="file-pdf"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 64C0 28.7 28.7 0 64 0H224V128c0 17.7 14.3 32 32 32H384V304H176c-35.3 0-64 28.7-64 64V512H64c-35.3 0-64-28.7-64-64V64zm384 64H256V0L384 128zM176 352h32c30.9 0 56 25.1 56 56s-25.1 56-56 56H192v32c0 8.8-7.2 16-16 16s-16-7.2-16-16V448 368c0-8.8 7.2-16 16-16zm32 80c13.3 0 24-10.7 24-24s-10.7-24-24-24H192v48h16zm96-80h32c26.5 0 48 21.5 48 48v64c0 26.5-21.5 48-48 48H304c-8.8 0-16-7.2-16-16V368c0-8.8 7.2-16 16-16zm32 128c8.8 0 16-7.2 16-16V400c0-8.8-7.2-16-16-16H320v96h16zm80-112c0-8.8 7.2-16 16-16h48c8.8 0 16 7.2 16 16s-7.2 16-16 16H448v32h32c8.8 0 16 7.2 16 16s-7.2 16-16 16H448v48c0 8.8-7.2 16-16 16s-16-7.2-16-16V432 368z"
              fill="currentColor"
            />
          </svg>
          View or Download PDF
        </a>
      </div>
    </div>
  </div>
</div>
`;
